qiskit/circuit/barrier.py

```diff
@@ -21,7 +21,10 @@
 
 
 class Barrier(Instruction):
-    """Barrier instruction."""
+    """A barrier is a visual indicator of the grouping of a circuit section. 
+    It also acts as a directive for circuit compilation to separate pieces 
+    of a circuit so that any optimizations or re-writes are constrained 
+    to only act between barriers."""
```

Comment:
I would turn this around, as they main aspect of a barrier is as a directive.

Reply:
Makes sense, is there anything I need to do to approve this change?

Reply:
nope.. I think we can merge after that